semialign/src/Data/Semialign/Internal.hs

```diff
@@ -742,7 +747,9 @@ instance Monad m => Semialign (Stream m) where
                 Done -> case (av, adone) of
                     (Just x, False) -> Yield (f $ This x) (sa, sb, Nothing, adone)
                     (_, True)       -> Done
+#if !MIN_VERSION_base(4,16,0)
                     _               -> Skip (sa, sb, Nothing, False)
+#endif
```

Comment:
Why did you need to do this? It compiles fine for me without making this change.

Reply:
Without this I get a warning from GHC about a redundant pattern match. Newer versions of GHC are smarter in recognizing those than previous versions.

Reply:
I'd rather match above with `av@(Just x)`

It wasn't obvious to me why the branch is redundant.